packages/jest-matcher-utils/src/index.js

```diff
@@ -110,8 +112,19 @@ const stringify = (object: any, maxDepth?: number = 10): string => {
     : result;
 };
 
-const printReceived = (object: any) => RECEIVED_COLOR(stringify(object));
-const printExpected = (value: any) => EXPECTED_COLOR(stringify(value));
+const highlightTrailingWhitespace = (text: string, bgColor: Function): string =>
+  text.split('\n')
```

Comment:
instead of splitting and joining after mapping, you can just use a global, multiline regular expression:
```
text.replace(/\s+$/gm, bgColor('$&'))
```